backend/worker/error_scorer.go

```diff
@@ -0,0 +1,283 @@
+package worker
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/model"
+	"github.com/highlight-run/highlight/backend/redis"
+	"github.com/highlight-run/highlight/backend/store"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type ErrorGroupWithLastObjectId struct {
+	ID                int
+	ProjectID         int
+	CreatedAt         time.Time
+	LastErrorObjectId int
+}
+
+type ErrorScorer struct {
+	store *store.Store
+	db    *gorm.DB
+	redis *redis.Client
+}
+
+func NewErrorScorer(store *store.Store, db *gorm.DB, redis *redis.Client) *ErrorScorer {
+	return &ErrorScorer{
+		store: store,
+		db:    db,
+		redis: redis,
+	}
+}
+
+func (errorScorer *ErrorScorer) ScoreImpactfulErrors(ctx context.Context) {
```

Comment:
roughly how long does this take? could we add some traces here to help with troubleshooting potential perf issues?

Reply:
Added some traces, and ran EXPLAIN ANALYZE below